enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,592 @@
+	// +kubebuilder:validation:Required
+	SignedPrefix string `json:"signedPrefix"`
+}
+
+const (
+	IdentityMatchPolicyMatchRepoDigestOrExact IdentityMatchPolicy = "MatchRepoDigestOrExact"
+	IdentityMatchPolicyMatchRepository        IdentityMatchPolicy = "MatchRepository"
+	IdentityMatchPolicyExactRepository        IdentityMatchPolicy = "ExactRepository"
+	IdentityMatchPolicyRemapIdentity          IdentityMatchPolicy = "RemapIdentity"
+)
+
+// +k8s:deepcopy-gen=true
+type ImagePolicyStatus struct {
+
+	// policyJSON contains the whole policy applied to the kubernetes namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
+
+	// conditions provide details on the status of this API Resource.
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Update container runtime config controller to watch ImagePolicy
+
+MCO container runtime config controller ImagePolicy syncHandler updates the signature verification configurations:
+- Retrieves all the ImagePolicy instances on the cluster.
+- Rre-validation: parse the images to validate the policy is not for scopes under OCP product repo. Report the error to machine config logs and the `status` field of that ImagePolicy CR.
+- Validates conflicting data from ImagePolicy instances. Ignore the policies from non global namespace CRs if the sigstore policy for the same image scope has been configured under global(openshift-config) namespace. Log this to the `status` field of both global and non global CRs.
+- Adds following configurations to a single machine config
+  - merges the instances for each namespace into a single [containers-policy.json](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) in a predefined `/path/to/policies/\<NAMESPACE\>.json`.
+  ImagePolicy CR should not change the `default` field of the cluster-wide policy.json. And the `default` of each  \<NAMESPACE\>.json will be the same as the cluster-wide `default` in /etc/containers/policy.json. The `default` might be `insecureAcceptAnything` or `reject`, the ImagePolicy handler needs to inherits the `default` value from Image CR handler.
+  - adds [/etc/containers/registries.d/*.yaml](https://github.com/containers/image/blob/main/docs/containers-registries.d.5.md) to allow matching sigstore signatures, for example:
+
+    ```yaml
+    docker:
+      my-registry/image:
+        use-sigstore-attachments: true
+    ```
+
+    Alternatively, we can enable it node wide:
+
+    ```yaml
+    default-docker:
+      use-sigstore-attachments: true
+    ```
+- The functions ImagePolicy syncHandler will be insert to current syncImageConfig() to coordinate with Image CR on editing `/etc/containers/policy.json`.
+- Merge rules for CRs and json string global policy: 
```

Comment:
Highlighting a comment from a “resolved” thread (which I can’t reopen):

The merge rules need to account for scopes nested inside cluster-admin-defined scopes: https://github.com/openshift/enhancements/pull/1402#discussion_r1343009236

Reply:
addressed https://github.com/openshift/enhancements/pull/1402/files#diff-ba23e083fdb5edb0f67de2d1b1053a4e6df0062dda73a1e255bbaf1f6337d5aaR278